src/com/moclojer/specs/moclojer.clj

```diff
@@ -28,61 +40,98 @@
-  (let [{:keys [content error?]} (render-template
-                                  (if-let [?external-body (:external-body response)]
-                                    (-> ?external-body
-                                        (enrich-external-body request)
-                                        ext-body/type-identification)
-                                    (:body response))
-                                  request)]
-    (try
+  (try
+    (let [{:keys [content error?]} (-> response
+                                       (get-response-body request)
+                                       (render-template request))]
       (cond
-        error? (let [parsed (json/read-str content :key-fn keyword)]
-                 {:error parsed
-                  :message (get parsed :message)})
-        (string? content) (try
-                            (json/read-str content)
-                            (catch Exception e
-                              (log/log :warn :json-parse-error
-                                       :body content
-                                       :message (.getMessage e))
-                              content))
-        :else content)
-      (catch Exception e
-        (log/log :error :bad-body
-                 :body content
-                 :message (.getMessage e))
-        {:error (.getMessage e)}))))
+        error?
+        (-> content
+            (json/read-str :key-fn keyword)
+            (as-> parsed {:error parsed
+                          :message (:message parsed)}))
+
+        (string? content)
+        (parse-json-safely content)
+
+        :else content))
+    (catch Exception e
+      (log/log :error :bad-body
+               :body (:body response)
+               :message (.getMessage e))
+      {:error (.getMessage e)})))
 
 (defn assoc-if [m k v]
   (if v
     (assoc m k v)
     m))
 
 (defn webhook-condition
-  "Checks `if` condition and return boolean value, default is true"
+  "Checks `if` condition and return boolean value, default is true when condition is empty"
   [condition request]
-  (let [template (str "{% if " condition " %}true{% else %}false{% endif %}")]
-    (if (empty? condition)
-      true
-      (boolean (Boolean/valueOf (selmer/render (ext-body/->str template) request))))))
+  (if (empty? condition)
+    true
+    (let [template (str "{% if " condition " %}true{% else %}false{% endif %}")]
+      (-> template
+          ext-body/->str
+          (selmer/render request)
+          Boolean/valueOf
+          boolean))))
 
 (defn build-parameters [request-values]
   (let [query (:query request-values)
         path (:path request-values)
         body (:body request-values)]
-    (-> (assoc-if {} :query-params query)
-        (assoc-if :path-params path)
-        (assoc-if :json-params body))))
+    (-> {}
+        (cond-> query (assoc :query-params query))
+        (cond-> path (assoc :path-params path))
+        (cond-> body (assoc :json-params body)))))
```

Comment:
There's an internal function, `assoc-if`. See if it's feasible using it. I like your solution though, never thought like that.

Reply:
```clojure
(defn assoc-some
  "Associates key-value pairs into a map, omitting any where the value is nil.
   Example:
     (assoc-some {:a 1} :b 2 :c nil)
     => {:a 1, :b 2}"
  [m & kvs]
  (reduce (fn [acc [k v]]
            (if (some? v)
              (assoc acc k v)
              acc))
          m
          (partition 2 kvs)))


(assoc-some {:a 1} :b 2 :c nil :d 4)
;; => {:a 1, :b 2, :d 4}
```

Reply:
migrate to `assoc-if`, the code became cleaner